src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -251,6 +251,7 @@ private void SetImplicitMetadataForCompileTimeAssemblies()
 
             foreach (var item in CompileTimeAssemblies)
             {
+                item.SetMetadata(MetadataKeys.NuGetSourceType, "Package");
```

Comment:
If we control all the consumers of this metadata, I'd prefer we just update them to check if there is a package ID set.

However, this change is also fine if you think it's better based on risk, churn, etc.